src/linter.rs

```diff
@@ -28,8 +28,19 @@ fn check_path(
         .iter()
         .any(|check_code| matches!(check_code.lint_source(), LintSource::AST))
     {
-        let python_ast = parser::parse_program(contents, "<filename>")?;
-        checks.extend(check_ast(&python_ast, contents, settings, autofix, path));
+        match parser::parse_program(contents, "<filename>") {
+            Ok(python_ast) => {
+                checks.extend(check_ast(&python_ast, contents, settings, autofix, path))
+            }
+            Err(parse_error) => {
+                if settings.select.contains(&CheckCode::E999) {
+                    checks.push(Check::new(
+                        CheckKind::SyntaxError(parse_error.error.to_string()),
```

Comment:
Wonder if we should use a different code here vs. in `main.rs`.

Reply:
Yeah, probably. Flake8 reports `E902` for filesystem errors.